docs(tcp): add documentation for k6/x/tcp extension and its API methods (#2233)
grafana/k6-docs: docs/sources/k6/next/javascript-api/k6-x-tcp/socket/on.md

## Events

### connect

Emitted when the socket successfully establishes a connection to the remote server.

#### Signature

```javascript
socket.on("connect", () => {
  // Connection established
})
```

### data

Emitted when data is received from the remote endpoint. The data is provided as a `Uint8Array`.

#### Signature

```javascript
socket.on("data", (data) => {
  // data is Uint8Array
})
```

#### Parameters

| Parameter | Type | Description |
| :-------- | :--- | :---------- |
| data | Uint8Array | The received data |

### close

Emitted when the socket connection is fully closed, either by the local or remote side. This is the final event in the socket lifecycle and is emitted exactly once per socket.

#### Signature

```javascript
socket.on("close", () => {
  // Connection fully closed
})
```

### error

Emitted when a socket error occurs, such as a connection failure or network issue.

#### Signature

```javascript
socket.on("error", (error) => {
  // Handle error
})
```

#### Parameters

| Parameter | Type | Description |
| :-------- | :--- | :---------- |
| error | Error | The error object |

### timeout

> Emitted when the socket times out due to inactivity as configured by `setTimeout()`. The connection is not automatically closed — call `destroy()` if you want to close it.

Reviewer: ```suggestion
Emitted when the socket times out due to inactivity as configured by `setTimeout()`. The socket doesn't automatically close the connection—call destroy() to close it.
```
